apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/internal/json/Utils.kt

```diff
@@ -62,9 +62,32 @@ object Utils {
 
       JsonReader.Token.BOOLEAN -> nextBoolean()
 
-      JsonReader.Token.NUMBER -> BigDecimal(nextString()!!)
+      JsonReader.Token.NUMBER -> {
+        val number = nextString()!!
+        // optimize
+        if (number.contains('.') || number.contains('e') || number.contains('E')) {
+          try {
+            number.toDouble()
+          } catch (e: Exception) {
+            number
+          }
+        } else {
+          try {
+            number.toInt()
+          } catch (e: Exception) {
+            number
```

Comment:
shouldn't we just fail here if it's not proper number?

Reply:
See above, a custom BigDecimal scalar adapter might want to read the number as a string